tokio/src/task/yield_now.rs

```diff
@@ -46,6 +46,15 @@ pub async fn yield_now() {
         type Output = ();
 
         fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<()> {
+            #[cfg(all(
+                tokio_unstable,
+                tokio_taskdump,
+                feature = "rt",
+                target_os = "linux",
+                any(target_arch = "aarch64", target_arch = "x86", target_arch = "x86_64")
+            ))]
+            crate::runtime::task::trace::Trace::leaf();
```

Comment:
This PR currently only marks `yield_now` as a leaf. That's fine, but I would like more futures to be included in the first release with taskdump support.

Since we just made a Tokio release, I think we can merge this now, since we will have plenty of time to add additional functions as leaves.